app/models/job_invocation_composer.rb

```diff
@@ -191,6 +263,21 @@ def compose
     self
   end
 
+  def trigger(raise_on_error = false)
+    generate_description
```

Comment:
Nice catch, the problem was with using the `pluck` method here https://github.com/theforeman/foreman_remote_execution/pull/149/files#diff-d3ff599b762e4ff95c9b2776d2be1079R159 - using the map instead (and working in memory) seems to fix the issue